portal/portal-impl/impl/src/java/org/sakaiproject/portal/charon/handlers/RoleSwitchHandler.java

```diff
@@ -55,12 +58,15 @@ public class RoleSwitchHandler extends BasePortalHandler
 	private String portalUrl;
 	private String externalRoles;
 
+	private ToolManager toolManager;
+
 	public RoleSwitchHandler() {
 		SpringBeanAutowiringSupport.processInjectionBasedOnCurrentContext(this);
 		setUrlFragment(RoleSwitchHandler.URL_FRAGMENT);
 		portalUrl = serverConfigurationService.getPortalUrl();
 		// get the roles that can be swapped to from sakai.properties
 		externalRoles = serverConfigurationService.getString("studentview.roles");
+		toolManager = ComponentManager.get(ToolManager.class);
```

Comment:
we should not be injecting a service with the ComponentManager when spring injection is occurring on the class.